main/modes/games/2048/mode_2048.c

```diff
@@ -0,0 +1,492 @@
+
+    // Handle inputs
+    buttonEvt_t evt;
+    switch (t48->state)
+    {
+        case T48_IN_GAME:
+        {
+            // Get inputs
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down)
+                {
+                    // Process inputs
+                    t48_gameInput(t48, evt.button);
+                }
+            }
+
+            // Loop the game
+            t48_gameLoop(t48, elapsedUs);
+            break;
+        }
+        case T48_START_SCREEN:
+        {
+            // Check any button is pressed
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down)
+                {
+                    soundPlaySfx(&t48->click, MIDI_SFX);
+                    t48_gameInit(t48);
+                    t48->state = T48_IN_GAME;
+                }
+            }
+            // Draw
+            t48_drawStartScreen(t48, t48_generateRainbow());
+            t48_chaseLEDs(elapsedUs);
+            break;
+        }
+        case T48_WIN_SCREEN:
+        {
+            // Get inputs
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down && (evt.button & PB_A || evt.button & PB_B))
+                {
+                    t48->state = T48_IN_GAME;
+                }
+            }
+            fillDisplayArea(0, 0, TFT_WIDTH, TFT_HEIGHT, c000);
+            drawText(&t48->titleFont, c055, youWin, (TFT_WIDTH - textWidth(&t48->titleFont, youWin)) / 2, 48);
+            drawText(&t48->font, c555, continueAB, (TFT_WIDTH - textWidth(&t48->font, continueAB)) / 2,
+                     TFT_HEIGHT - 64);
+            break;
+        }
+        case T48_HS_SCREEN:
+        {
+            // Handle input in text entry object
+            while (checkButtonQueueWrapper(&evt))
+            {
+                t48->textEntryDone = !textEntryInput(evt.down, evt.button);
+            }
+
+            // If the text entry is done, sort the scores, reset the text entry object,
+            // and display the final screen
+            if (t48->textEntryDone)
+            {
+                t48SortHighScores();
+                textEntrySoftReset();
+                t48->state = T48_END_SCREEN;
+            }
+
+            // Draw text entry
+            textEntryDraw(elapsedUs);
+            break;
+        }
+        case T48_END_SCREEN:
+        {
+            // Check any button is pressed
+            while (checkButtonQueueWrapper(&evt))
+            {
+                if (evt.down && (evt.button & PB_A || evt.button & PB_B))
+                {
+                    soundPlaySfx(&t48->click, MIDI_SFX);
+                    t48->state = T48_START_SCREEN;
+                }
+            }
+
+            // Draw the final score screen
+            t48_drawGameOverScreen(t48, t48->score, t48_generateRainbow());
+            t48_chaseLEDs(elapsedUs);
+            break;
+        }
+        default:
+        {
+            break;
```

Comment:
If for some reason the game state becomes one of the non-enum values, the swadge will softlock here because the button won't be read to return to the main menu. If the state value is invalid we've got bigger problems, so this is more of an interesting note than something that needs to be tweaked.

Reply:
I mean, we can move it to the start screen then. It's not a hard line to move.

But yes, I get what you're saying.